fe/fe-core/src/main/java/org/apache/doris/alter/InvertedIndexJob.java

```diff
@@ -0,0 +1,400 @@
+                long originTabletId = originTablet.getId();
+                List<Replica> originReplicas = originTablet.getReplicas();
+                for (Replica originReplica : originReplicas) {
+                    if (originReplica.getBackendId() < 0) {
+                        LOG.warn("replica:{}, backendId: {}", originReplica, originReplica.getBackendId());
+                        throw new AlterCancelException("originReplica:" + originReplica.getId()
+                                + " backendId < 0");
+                    }
+                    AlterInvertedIndexTask alterInvertedIndexTask = new AlterInvertedIndexTask(
+                            originReplica.getBackendId(), db.getId(), olapTable.getId(),
+                            partitionId, originIndexId, originTabletId,
+                            originSchemaHash, olapTable.getIndexes(),
+                            alterInvertedIndexes, originSchemaColumns,
+                            isDropOp, taskSignature);
+                    invertedIndexBatchTask.addTask(alterInvertedIndexTask);
+                }
+            } // end for tablet
+
+            LOG.info("invertedIndexBatchTask:{}", invertedIndexBatchTask);
+            AgentTaskQueue.addBatchTask(invertedIndexBatchTask);
+            AgentTaskExecutor.submit(invertedIndexBatchTask);
+            // partitionInvertedIndexTaskMap.put(partitionId, invertedIndexBatchTask)
+        } finally {
+            olapTable.readUnlock();
+        }
+        this.jobState = JobState.RUNNING;
+        LOG.info("transfer inverted index job {} state to {}", jobId, this.jobState);
+    }
+
+    protected void runRunningJob() throws AlterCancelException {
+        Preconditions.checkState(jobState == JobState.RUNNING, jobState);
+
+        if (!invertedIndexBatchTask.isFinished()) {
+            LOG.info("inverted index tasks not finished. job: {}, partitionId: {}", jobId, partitionId);
+            // TODO: task failed limit
+            return;
+        }
+
+        this.jobState = JobState.FINISHED;
+        this.finishedTimeMs = System.currentTimeMillis();
+
+        Env.getCurrentEnv().getEditLog().logInvertedIndexJob(this);
+        LOG.info("inverted index job finished: {}", jobId);
+    }
+
+    protected boolean cancelImpl(String errMsg) {
+        return true;
+    }
+
+    public void replay(InvertedIndexJob replayedJob) {
+        try {
+            InvertedIndexJob replayedInvertedIndexJob = (InvertedIndexJob) replayedJob;
+            switch (replayedJob.jobState) {
+                case WAITING_TXN:
+                    replayCreateJob(replayedInvertedIndexJob);
+                    break;
+                case FINISHED:
+                    replayRunningJob(replayedInvertedIndexJob);
+                    break;
+                case CANCELLED:
+                    // TODO:
+                    // replayCancelled(replayedInvertedIndexJob);
+                    break;
+                default:
```

Comment:
What about RUNNING state?

Reply:
RUNNING state do not write edit log, so no need replay it